File: Python/Count Primes.py
class Solution:
    # @param {integer} n
    # @return {integer}
    def countPrimes(self, n):
        if n < 2:
            return 0
        listPrime=[True] * n
        listPrime[0] = listPrime[1] = False
        for i in range(2, int(n**0.5) + 1):
            if listPrime[i]:
                listPrime[i*i:n:i] = [False] * len(listPrime[i*i:n:i])
        return listPrime.count(True)

File: java/Count_Primes.java
// Prime method : Filter Method！------Sieve of Eratosthenes : https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes
// Time O(n*loglogn) , space O(n).

public class Solution {
    public int countPrimes(int n) {
        boolean[] not = new boolean[n];
        Arrays.fill(not, false);
        int count = 0;
        for (int i = 2; i < n; ++i) {
            if (not[i]) continue;
            count++;
            int k = i*2;
            while (k < n) {
                not[k] = true;
                k += i;
            }
        }
        return count;
    }
}
